### PR TITLE
CA-315952 fix JSON output

### PR DESCRIPTION
**Pull request by lindig**
Json so far is generated in a naive way that violates the Json
specification by ignoring basic quoting rules. This commit delegates
generation of Json to a library instead and should be much more
declarative. Furthermore, it emits data points that previously were hard
coded in the Json output.

### CODE REVIEW

File: lib/rrd.ml [hunk 1]
```diff
@@ -692,55 +692,76 @@ let xml_to_output rrd output =
...
+
+  let datasource ds =
+    record
+      [ "name"               , string "%s" ds.ds_name
+      ; "type"               , ty ds.ds_ty
+      ; "minimal_hearbeat"   , string "%s" (Utils.f_to_s ds.ds_mrhb)
+      ; "min"                , string "%s" (Utils.f_to_s ds.ds_min)
+      ; "max"                , string "%s" (Utils.f_to_s ds.ds_max)
+      ; "last_ds",           ( match ds.ds_last with
+                             | VT_Float x -> string "%s" (Utils.f_to_s x)
```
Comment (edwintorok) [inline lib/rrd.ml]: Here is an example RRD json: https://github.com/oetiker/rrdtool-1.x/blob/f4040b2c1b9e51fe23b8bc546b43e0b33440f855/tests/xport1.json.output

Comment (edwintorok) [reply lib/rrd.ml]: We can defer this to a future PR

File: lib/rrd.ml [hunk 2]
```diff
@@ -692,55 +692,72 @@ let xml_to_output rrd output =
...
-        "},database:"^(do_database rra.rra_data)) rra_list))^"}]"
-  in
+      array @@ Array.to_list @@ Array.init rows (fun row ->
+          array @@ Array.to_list @@ Array.init cols (fun col -> get rings rows row col))
+
+  let rra x =
+    record
+      [ "cf"           , string "%s" (cf_type_to_string x.rra_cf)
+      ; "pdp_per_row"  , string "%d" x.rra_pdp_cnt
+      ; "params"       , string "%s" (Utils.f_to_s x.rra_xff)
```
Comment (psafont) [inline lib/rrd.ml]: Previously this was more deeply nested as `params: {xff: rra_xff_value}`, was this changed on purpose?

Comment (lindig) [reply lib/rrd.ml]: This wasn't on purpose. Will take look.

**Pull request merged**